packages/block-editor/src/components/block-list/index.js

```diff
@@ -46,16 +51,47 @@ function BlockList(
 			isDraggingBlocks,
 		} = select( 'core/block-editor' );
 
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const multiSelectedBlockClientIds = getMultiSelectedBlockClientIds();
+
+		// Determine if there is an active template part (or other entity) to spotlight.
+		// Restrict to Full Site Editing experiment.
+		const isFullSiteEditingEnabled = getSettings()
+			.__experimentalEnableFullSiteEditing;
+		const activeEntityBlockId =
+			isFullSiteEditingEnabled &&
+			( function () {
+				// Check if selected block is a valid entity area.
+				if (
+					ENTITY_AREAS.includes(
+						getBlockName( selectedBlockClientId )
+					)
+				) {
+					return selectedBlockClientId;
+				}
+				// Check if first selected block is a child of a valid entity area.
+				const entityAreaParents = getBlockParentsByBlockName(
+					selectedBlockClientId || multiSelectedBlockClientIds[ 0 ],
+					ENTITY_AREAS
+				);
+				if ( entityAreaParents ) {
+					// Last parent closest/most interior.
+					return last( entityAreaParents );
+				}
+				return null;
+			} )();
```

Comment:
This is a bit too complex for `mapSelectToProps`, should this be moved to a selector?

Reply:
I was just thinking this as well last night, that makes sense to me.

Reply:
Also noting moving this logic to a selector will allow us to clean up the document-actions header component in edit-site since it could use this information as well.

Reply:
updated